zanata-war/src/main/java/org/zanata/search/HTextFlowTargetIndexingStrategy.java

```diff
@@ -60,4 +68,62 @@ protected ScrollableResults queryResults(int ignoredOffset, FullTextSession sess
         query.setFetchSize(Integer.MIN_VALUE);
         return query.scroll(ScrollMode.FORWARD_ONLY);
     }
+
+    public void reindexForProject(HProject project, FullTextSession session,
+            AsyncTaskHandle handle) {
+        // it must use the same session in the DAO and to do the indexing
+        HTextFlowTargetStreamingDAO dao =
+                new HTextFlowTargetStreamingDAO(HTextFlowTarget.class, session);
+        ScrollableResults scrollableResults =
+                dao.getTargetsWithAllFieldsEagerlyFetchedForProject(
+                        project);
+        reindexScrollableResultSet(session, scrollableResults, handle);
+    }
+
+    private static void reindexScrollableResultSet(FullTextSession session,
```

Comment:
In order for this method to be asynchronous it needs to be annotated with `@Async` and the Seam component needs to be annotated with `@ContainsAsyncMethods`. The method can return a `java.util.concurrent.Future` or void. For an example, take a look at `AsyncTaskITCase`